### DISCUSSION TITLE
Issue with Argo workflow using helm install

### DISCUSSION THREAD

**Discussion by mysandy603** [argoproj/argo-helm#1705] [Q&A]
Hello, 

We are trying to deploy Argo Workflow template and Argo workflows manifest that under template directory using helm install command.
But when we trigger the helm install, we notice that the Workflows manifest is created before workflowTemplate. and hence workflow is failed for the first time with the error that "Workflow template was not found."
When we retrigger the workflow again, it is successfully able to find the workflow template and we are able to successfully run the workflow.
We are using Argo 3.3.9 version. Can any one help for this resolution.

Thanks,
Sandeep

**Comment by yu-croco** [comment]
Hi @mysandy603 ,
in order to manage the orders of applying k8s resource, I think [Chart Hooks](https://helm.sh/docs/topics/charts_hooks/) would help you in your case. 🤔
